docs/glossary.md

```diff
@@ -0,0 +1,208 @@
+# Glossary
+
+## Anonymous branch
+
+An anonymous branch is a chain of commits that doesn't have any
+[named branches](#branch) pointing to it or to any of its descendants. Unlike
+Git, Jujutsu keeps commits on anonymous branches around until they are
+explicitly abandoned. Visible anonymous branches are tracked by the
+[view](#view), which stores a list of [heads](#head) of such branches.
+
+## Backend
+
+A backend is an implementation of the storage layer. There are currently two
+builtin commit backends: the Git backend and the native backend. The Git backend
+stores commits in a Git repository. The native backend is used for testing
+purposes only. Alternative backends could be used, for example, if somebody
+wanted to use jj with a humongous monorepo (as Google does).
+
+There are also pluggable backends for storing other information than commits,
+such as the "operation store backend" for storing
+[the operation log](#operation-log).
+
+## Branch
+
+A branch is a named pointer to a [commit](#commit). They automatically follow
+the commit if it gets [rewritten](#rewrite). Branches are sometimes called
+"named branches" to distinguish them from
+[anonymous branches](#anonymous-branch), but note that they are more similar
+to Git's branches than to
+[Mercurial's named branches](https://www.mercurial-scm.org/wiki/Branch#Named_branches).
+See [here](branches.md) for details.
+
+## Change
+
+A change is a commit as it [evolves over time](#rewrite).
+
+## Change ID
+
+A change ID is a unique identifier for a [change](#change). They are typically
+16 bytes long and are often randomly generated. By default, jj log presents
+them as a sequence of 12 letters in the k-z range, at the beginning of a line.
+These are actually hexadecimal numbers that use "digits" k-z instead of 0-9a-f.
```

Comment:
Very minor and optional: on second thought, perhaps it makes sense to write the range as "z-k" here to hint at correspondence.

Reply:
Done (here, but I left it as "k-z" on the line above since order seems unimportant there)